contrib/etcd/pkg/apis/sources/v1alpha1/etcdsource_types.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EtcdSource is the Schema for the etcdsources API
+// +k8s:openapi-gen=true
+type EtcdSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EtcdSourceSpec   `json:"spec,omitempty"`
+	Status EtcdSourceStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EtcdSourceList contains a list of EtcdSource
+type EtcdSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []EtcdSource `json:"items"`
+}
+
+const (
+	// EtcdConditionReady has status True when the EtcdSource is ready to send events.
+	EtcdConditionReady = duckv1alpha1.ConditionReady
+
+	// EtcdConditionSinkProvided has status True when the EtcdSource has been configured with a sink target.
+	EtcdConditionSinkProvided duckv1alpha1.ConditionType = "SinkProvided"
+
+	// EtcdConditionDeployed has status True when the EtcdSource has had it's deployment created.
+	EtcdConditionDeployed duckv1alpha1.ConditionType = "Deployed"
+)
+
+var etcdCondSet = duckv1alpha1.NewLivingConditionSet(
+	EtcdConditionSinkProvided,
+	EtcdConditionDeployed,
+)
+
+// EtcdSourceSpec defines the desired state of EtcdSource
+type EtcdSourceSpec struct {
+	// Resources is the list of resources to watch
```

Comment:
ooohhhh ok, I think this source is misnamed. It is not etcd you are sourcing, it is the k8s api server.